sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -71,39 +67,14 @@ def parse_connection_string(conn_str):
     :type conn_str: str
     :rtype: ~azure.servicebus.ServiceBusConnectionStringProperties
     """
-    conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
-    if any(len(tup) != 2 for tup in conn_settings):
-        raise ValueError("Connection string is either blank or malformed.")
-    conn_settings = dict(conn_settings)
-    shared_access_signature = None
-    for key, value in conn_settings.items():
-        if key.lower() == "sharedaccesssignature":
-            shared_access_signature = value
-    shared_access_key = conn_settings.get("SharedAccessKey")
-    shared_access_key_name = conn_settings.get("SharedAccessKeyName")
```

Comment:
the connection string parser actually cares about the case of the string while the `_parse_conn_str`does not: e.g. `sharedaccesskey` vs `SharedAccessKey`.

so we need to take this into account when combining the parse logic.

Reply:
also I think it's worth to add tests against the case of the string in the two scenarios